File: app/src/main/java/com/example/sunnyenterprise/activities/SplashActivity.java
package com.example.sunnyenterprise.activities;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.os.Handler;
import android.preference.PreferenceManager;
import android.view.Window;
import android.view.WindowManager;
import android.widget.Toast;

import com.example.sunnyenterprise.R;

public class SplashActivity extends AppCompatActivity {

    int SPLASH_TIME = 3000;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
        boolean Islogin = prefs.getBoolean("Islogin", false);

        if(Islogin){
            Intent i = new Intent(SplashActivity.this, HomeActivity.class);
            startActivityForResult(i, 1);
        }

        else{
            new Handler().postDelayed(() -> {
                Intent loginIntent = new Intent(SplashActivity.this, LoginActivity.class);
                startActivity(loginIntent);
                finish();

            }, SPLASH_TIME);
        }



    }
}

File: app/src/main/java/com/example/sunnyenterprise/activities/SplashActivity.kt
package com.example.sunnyenterprise.activities

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.preference.PreferenceManager
import android.util.Log
import android.view.Window
import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity
import com.example.sunnyenterprise.R
import com.example.sunnyenterprise.utils.Preferences

@Suppress("DEPRECATION")
class SplashActivity : AppCompatActivity() {
    var preferences: Preferences? = null

    var SPLASH_TIME = 3000
    override fun onCreate(savedInstanceState: Bundle?) {
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        preferences = Preferences(this)

        Handler().postDelayed({
            if (preferences!!.getMobile().isEmpty()) {
                val loginIntent = Intent(this@SplashActivity, LoginActivity::class.java)
                startActivity(loginIntent)
                finish()
            } else {
                Log.d("ids", "customer Id : ${preferences!!.getId()}")
                val homeIntent = Intent(this@SplashActivity, HomeActivity::class.java)
                startActivity(homeIntent)
                finish()
            }
        }, SPLASH_TIME.toLong())

    }
}